src/main/java/org/sagebionetworks/web/client/Portal.java

```diff
@@ -47,13 +47,14 @@ public static PortalGinInjector getInjector() {
   public void onModuleLoad() {
     zeroOpacity(RootPanel.get("headerPanel"), RootPanel.get("rootPanel"));
     detectProxiedWebsiteAttack();
-    // we might need to reload using the new token scheme (required for SEO)
+    // Test to see if client is instructed to load via "#!" fragment.
+    // If detected, reload using a new path-based token scheme (required for SEO).
+    // The GWT History class assumes tokens begin with '#', and we used to include the '!' as the prefix for all GWT Places.
     String initToken = History.getToken();
-    if (initToken.length() > 0 && !initToken.startsWith("!")) {
+    if (initToken.length() > 0 && initToken.startsWith("!")) {
       String fullUrl = Window.Location.getHref();
-      fullUrl = fullUrl.replace("#" + initToken, "#!" + initToken);
-      Window.Location.replace(fullUrl);
-      Window.Location.reload();
+      fullUrl = fullUrl.replace("#!", "");
+      Window.Location.assign(fullUrl);
```

Comment:
This is important - this is the client-side check for #! during app initialization.  If detected, will remove the #! and reload the browser.